python/kserve/kserve/model.py

```diff
@@ -75,19 +90,43 @@ def __init__(self, name: str):
         self._http_client_instance = None
         self._grpc_client_stub = None
 
-    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR):
-        request = await self.preprocess(body) if inspect.iscoroutinefunction(self.preprocess) \
-            else self.preprocess(body)
+    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR, headers={}):
```

Comment:
there is a PR that will pass in headers here that is IP https://github.com/kserve/kserve/pull/2360

one question - if there is no header (like, if you aren't using queue proxy) do we want to create a request id or leave it as none?